From 8a0b7319dfaac08b45106a4062c9856860e50de4 Mon Sep 17 00:00:00 2001
From: Philip Prindeville <philipp@redfish-solutions.com>
Date: Tue, 10 Apr 2018 16:57:06 -0600
Subject: [PATCH] env: only use color diffs on terminals

When sending script/env diff's output to a file or pipeline, you
don't want escape characters for ANSI color sequences fouling the
byte stream.

Signed-off-by: Philip Prindeville <philipp@redfish-solutions.com>
---
 scripts/env | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/scripts/env b/scripts/env
index 63e26eb154d..f7fdccf6825 100755
--- a/scripts/env
+++ b/scripts/env
@@ -104,7 +104,11 @@ env_list() {
 env_diff() {
 	env_init
 	env_sync_data
-	git diff --cached --color
+	if [ -t 1 ]; then
+		git diff --cached --color
+	else
+		git diff --cached
+	fi
 	env_link_config
 }
 
